src/components/VTextField/VTextField.js

```diff
@@ -84,12 +84,17 @@ export default {
     },
     inputValue: {
       get () {
-        return this.value
+        return this.lazyValue
       },
       set (val) {
-        this.lazyValue = val
-        this.mask && this.setSelectionRange()
-        this.$emit('input', val)
+        // Inner unmaskText() strips off non-alphanum
+        // maskText() masks and removes dirty alphanum
+        // Outer unmaskText() provides a clean lazyValue
+        const value = this.unmaskText(this.maskText(this.unmaskText(val)))
+        this.lazyValue = typeof val === 'number' ? +value : value
+        this.mask ? this.setSelectionRange()
+          : this.$emit('input', this.returnMaskedValue
+            ? this.$refs.input.value : this.lazyValue)
```

Comment:
Why are we returning different values based on `returnMaskedValue` when there is no mask?

Reply:
Also, this whole function should probably be split by an if statement so none of this casting is happening on non-masked inputs. The same applies for the value watcher too.